### DISCUSSION TITLE
terrible performance on the rust plugin/intellij CE

### DISCUSSION THREAD

**Discussion by petar-dambovaliev** [intellij-rust/intellij-rust#8353] [Q&A]
I am on the latest version of both the plugin and the IDE.
I am on a brand new top notch macbook pro with 32gb ram and a monster CPU.
I have given the IDE 4gb of ram for usage.
I have enabled code analysis on file save.
As soon as i have a source file that is 5k lines, i have to wait 30-40 seconds for the analysis to be done (cargo check).
Also, sometimes it doesn't even update the shown errors.
I would need to make another change to the file (a 3rd one) to trigger, for example, the removal of the underlining red for errors.
I need to wait 10-15 seconds for autocomplete.
This to me is unacceptable.

**Comment by vlad20012** [comment]
Hi!
Are you working on an open-source project? I'd love to see a project that leads to performance issues like this.

> As soon as i have a source file that is 5k lines, i have to wait 30-40 seconds for the analysis to be done (cargo check).

We can't do anything with `cargo check` itself, unfortunately :( I can suggest disabling `cargo check on the fly` and running it manually

> I have enabled code analysis on file save.

What is "code analysis on file save"?

**Reply by petar-dambovaliev** [reply]
I am not working on an open source project and there is no way i can share the code, as it's owned by my company.
I think, the problem is a product of a large file with a lot of dependencies.
We have about 800 dependencies in the project.
`cargo check` works fast, when i do it in the terminal.
What is slow is the response of the plugin/IDE to update the state(show errors or remove them) when i update a file.
`code analysis` refers to the checkbox `Run external linter to analyze code on the fly`.

**Comment by jbabyhacker** [comment]
I have noticed this as well. I am running CLion on Ubuntu 20.04 with 32 core CPU. My codebase is about 12k lines. I have tried previous versions, v161, v162, v163 and didn't notice a change between them, but with v164, there is a significant regression in performance. I have reverted to v163 and will stay here until the performance issue is fixed. In v164, everything is slower: syntax coloring, auto-complete, hints, CPU usage is extremely high. You can obtain older versions from here to see if this fixes the issue. [https://plugins.jetbrains.com/plugin/8182-rust/versions/stable](https://plugins.jetbrains.com/plugin/8182-rust/versions/stable)

**Reply by Undin** [reply]
@dima74 could you take a look, please?

**Reply by dima74** [reply]
@jbabyhacker Hi! Sorry that you experience performance problems with latest update. Could you please capture CPU and memory snapshot as described [here](https://intellij-support.jetbrains.com/hc/en-us/articles/207241235-Reporting-performance-problems). Also screenshot from [activity monitor](https://www.jetbrains.com/help/idea/activity-monitor.html)   would be helpful

**Reply by jbabyhacker** [reply]
@dima74 I have been running with v165 today and the problems I observed in v164 are not in this release. I don't think there is anything else we need to do.
